### DISCUSSION TITLE
No JSON Schema generator from database schemas?

### DISCUSSION THREAD

**Discussion by jeffreytgilbert** [json-schema-org/community#524] [General]
Validating forms is annoying, repetitive, redundant, error prone work that’s duplicated effort on the client, server, and then the database, not to mention in test code on both the server and client. If there is no json schema generator which takes a database connection or schema and outputs a json schema file for use on both the client and server, then why not? Surely someone somewhere has written a library for doing this.

**Comment by gregsdennis** [comment]
Have you looked on our implementations page?  There are several listed in various languages.

https://json-schema.org/implementations#schema-generators
